bpo-42128: Add documentation for pattern matching (PEP 634) (#24664)
python/cpython: Doc/tutorial/controlflow.rst

:keyword:`!elif` ... sequence is a substitute for the ``switch`` or
``case`` statements found in other languages.

> If you're making comparison with constants, or checking for specific types or

Reviewer: 1. "making comparison with" isn't a great phrase (I'd use the plural, "comparisons", or say "comparing to")
2. The match statement is really only useful if *the same* variable is used in each condition

```suggestion
If you're comparing the same value to several constants, or checking for specific types or
```

   ...     pass   # Remember to implement this!
   ...

> .. _tut-match:

Reviewer: Note that we now have two slightly diverging versions of the same introduction to match statements in the docs (three if you count Appendix A of PEP 636) -- the other is in what's new in 3.10. We don't have to fix this now but I think it would be good if the texts didn't arbitrarily differ from each other. Either the same text should be included in both places (possibly with a different intro and outro), or what's new should be much shorter and just point here. Again, doesn't need to be addressed today, but something to keep in mind. /CC @willingc

Reviewer 2: It makes good sense to bring the two closer together. I would err on the side of clarity over length in the What's New as it's what most folks will read before or after the PEP tutorial.
